src/presign/participant.rs

```diff
@@ -486,13 +484,15 @@ impl PresignParticipant {
             &self.storage,
             StorableType::PresignRoundTwoPrivate,
             message.id(),
-        )?;
+        )
+        .is_ok();
         let all_publics_received = has_collected_all_of_others(
             &self.other_participant_ids,
             &self.storage,
             StorableType::PresignRoundTwoPublic,
             message.id(),
-        )?;
+        )
+        .is_ok();
         if all_privates_received && all_publics_received {
```

Comment:
You new code here does something slightly different from the original. In the original version we could return early with a Result<...> if the `?` encountered an error. The way the current code is written we never return early here. We always execute both `has_collected_all_of_others` checks and only return Ok(...).

I'm not sure if this is the intended behavior and if it is correct. Just wanted to point it out. What do you think?

Reply:
I guess this distinction matters if `has_collected_all_of_others` can throw an error besides `StorageItemNotFound`. The only other possible error is in the `serialize!` macro which I don't think should be an issue. Some `StorageItemNotFound` errors are expected in general execution and shouldn't be propagated all the way to the top, so there has to be some code at some point that turns it into "not an error", right? 

Though this seems like a situation where distinguishing between expected errors and more serious ones will matter in the long term.

Reply:
Maybe this is a reason to just have Storage-related queries return bools rather than errors?

Reply:
> Maybe this is a reason to just have Storage-related queries return bools rather than errors? Then we can save errors for "something else went wrong" like unable to access storage or something

Right. That's the main disadvantage of having our errors mean something different in this context. Okay, to avoid having to think about what the error means for specific functions, I think we should change it so these functions return a `Result<bool, InternalError>` (I think that's the right type? This way we can always use `?` on functions and check the bool value directly.

In the future these functions may be able to return other types of 'unrecoverable` errors. (Maybe not but it is good to have our code be robust to changes like these).

Reply:
I went ahead an changed functions like `contains_batch` and `is_keygen_done` to return `Result<bool, InternalError>`.